sec2_the_ruby_programming_language/2.3_number.rb

```diff
@@ -0,0 +1,75 @@
+# Numbers
+puts 1 + 2       #3
+puts 2 - 1       #1
+puts 10 * 2      #20
+puts 10 / 4      #2
+puts 10.0 / 4    #2.5
```

Comment:
why these happen here?

Reply:
This is because the answer will follow the data type of the left operand of the arithmetic operator.

When the integer is divided by an integer, the end result will be an integer.
When the float number is divided by an integer, the end result will be a float number.

However, when both operands have the same data type then the end result will also have the same data type as both operands.

Reply:
This is because many binary operators that expect operands of arithmetic type cause conversions and yield result types in a similar way. This pattern is called the usual arithmetic conversions. 

In this case, when the float number is divided by an integer, the end result will also be converted into a float number automatically.

Reply:
Speak english please

Reply:
Just explain like you explain to your grandma will do😂 @zi-qiang

Reply:
10 / 4 = 2 in line 5 because it is an integer where there is no decimal places tailing behind.
10.0 /4 = 2.5 in line 6 because it is a float number

Reply:
I can see that. But I don't understand why. My question is why.
Logically with maths, 10/4 == 2.5. Why in this case it isn't?

Reply:
Ya, the answer will not have any decimal places unless we convert it to become a float number. Else, it will become the answer will assigned the data type same as the operands.

Reply:
> the answer will not have any decimal places unless we convert it to become a float number

Based on what you say, unless we convert `it` to become a float number. Thats saying it should be correct if we convert the answer into decimal place right? Which is [this](https://github.com/ziqiang2344/Learn_Ruby_on_Rails/pull/4#discussion_r846716080). But, this seems to show differently.

I really confused with your explanation

Reply:
So the question here is asking why 10/4 is equal to 2 instead of 2.5. 

In this case, 10 & 4 are both integer numbers, which also means that, after the computation, `10/4` will be an integer by default also. Hence, the answer will be a `2`

However, if we put convert one or both of the operands into a float number. For example, we convert `10` into `10.0` / `4` into `4.0` / 10 & 4 into `10.0` and `4.0`. Then, all of it will result in `2.5`.

In summary,
1. integer / integer = integer
2. integer /float = float
3. float/integer = float
4. float/float  = float

Reply:
Ok. But, what I have been asking from the beginning is why `integer / integer = integer`

Reply:
This is because it is being specified in the [Ruby spec](https://github.com/ruby/spec/blob/master/core/integer/divide_spec.rb#L19-L21) that when an integer is divided by an integer, an integer division will occur and hence it will return an integer answer as well.